### DISCUSSION TITLE
How to disable native keyboard from showing up on Android and iOS?

### DISCUSSION THREAD

**Discussion by DoubleDBE** [AvaloniaUI/Avalonia#14274] [Q&A]
In my application, I created my own simple keyboard in XAML.

When a TextBox gets focused, naturally the native keyboard is shown.

But this is not the desired behavior for my use case.

Is there a way I can disable this on a global scale?

I already tried adding `android:windowSoftInputMode="stateAlwaysHidden"` to the android manifest, but it doesn't work.

I'm using the latest version of Avalonia (11.0.7).

Some insights would be greatly appreciated!

**Comment by thevortexcloud** [comment]
This might help you?

https://github.com/AvaloniaUI/Avalonia/pull/13363

Not sure though.

**Reply by DoubleDBE** [reply]
Well, if i'd only knew what to look for...

Would be cool if they added `TextInputOptions.ContentType="Disabled"` or something.

**Reply by thevortexcloud** [reply]
I would suggest putting a feature request. There is a lot of work happening in this area already from what I have seen.

**Comment by kekekeks** [comment]
For now you can handle `TextInputMethodClientRequestedEventArgs` on the view level (`Bubble` stage, `handledEventsToo: true`) and reset whatever input client that was set by the textbox. That will make our input system to think that the focused control doesn't have text input capabilities.

**Reply by DoubleDBE** [reply]
Sounds good for now! How would this look like in code?

**Reply by kekekeks** [reply]
`control.AddHandler(InputElement.TextInputMethodClientRequested, ...)`

**Reply by DoubleDBE** [reply]
Thanks, still not sure where and how to apply it, and what to "reset" though? Sorry, newbie Avalonia user here.

**Reply by workgroupengineering** [reply]
`control.RemoveHandler(InputElement.TextInputMethodClientRequested, ...)`

**Reply by beemobile4net** [reply]
I am facing the same problem on Android. I followed the advice.
This part works - it properly registers an event handler (`txtMain` is in my case reference to Avalonia TextBox):
``` csharp
txtMain.AddHandler(InputElement.TextInputMethodClientRequestedEvent, this.TextMain_InputMethodClientRequested, RoutingStrategies.Bubble, true);
```

However, I'm struggling how to implement the `TextMain_InputMethodClientRequested` method:

``` csharp
private void TextMain_InputMethodClientRequested(object? sender, TextInputMethodClientRequestedEventArgs e)
{
    TextInputMethodClient? timc = e.Client; // TextInputMethodClient object successfully retrieved
    // how to hide the the client now? There is no such method on TextInputMethodClient
}
```

I do not know, how to hide the soft input method by using a reference to TextInputMethodClient. I do not see any method, nor property, that I could use. I'm on Avalonia 11.1.0-rc1.

Thank you.

**Reply by DoubleDBE** [reply]
You should set e.Client to null, works for me;

e.Client = null;

@kekekeks is this still applicable, doing it like this?

**Reply by beemobile4net** [reply]
@DoubleDBE  - thank you for a fast response. I will try it out once I get to the office again. I will definitely write a comment about my findings.

**Reply by beemobile4net** [reply]
The reason why I did not report on my findings was, that the advice did not work for me. Then I had to switch to a different project. But the issue continues to be unresolved for me.

**Comment by davidhollitervis** [comment]
Is there a easy way or has anyone a code example how to implement it for certain controls?

We currently fork the textbox and textboxinputmethod class to solve the problem but every update is a bit annoying.
